src/connector.js

```diff
@@ -141,417 +92,654 @@ let pageTemplateHTML = document.getElementById( 'sr-pager-page-' )?.innerHTML;
+			}
+			else if( fqupdate === "datemodified_dt:[now-7days to now]" ) {
+				aqString += ' @date>today-7d';
+			}
+			else if( fqupdate === "datemodified_dt:[now-1month to now]" ) {
+				aqString += ' @date>today-30d';
+			}
+			else if( fqupdate === "datemodified_dt:[now-1year to now]" ) {
+				aqString += ' @date>today-365d';
+			}
 		}
-	}
-	
-	if ( urlParams.fqupdate ) {
-		let fqupdate = urlParams.fqupdate.toLowerCase();
-		if ( fqupdate === "datemodified_dt:[now-1day to now]" ) {
-			aqString += ' @date>today-1d';
+		if ( urlParams.dmn ) {
+			aqString += ' @hostname="' + urlParams.dmn + '"';
 		}
-		else if( fqupdate === "datemodified_dt:[now-7days to now]" ) {
-			aqString += ' @date>today-7d';
+
+		if ( urlParams.sort ) {
+			const sortAction = loadSortCriteriaActions( headlessEngine ).registerSortCriterion( {
+				by: "date",
+				order: "descending",
+			} );
+			headlessEngine.dispatch( sortAction );
 		}
-		else if( fqupdate === "datemodified_dt:[now-1month to now]" ) {
-			aqString += ' @date>today-30d';
+
+		if ( urlParams.elctn_cat ) {
+			let elctn_cat = urlParams.elctn_cat.toLowerCase();
+			if( elctn_cat === "his" ) {
+				aqString += ' @uri="dir=his"';
+			}
+			else if( elctn_cat === "comp" ) {
+				aqString += ' @uri="compendium"';
+			}
+			else if( elctn_cat === "ogi" ) {
+				aqString += ' @uri="dir=gui"';
+			}
+			else if( elctn_cat === "officer_manuals" ) {
+				aqString += ' @uri="dir=pub"';
+			}
+			else if( elctn_cat === "research" ) {
+				aqString += ' @uri="dir=rec"';
+			}
+			else if( elctn_cat === "press_release" ) {
+				aqString += ' @uri="dir=pre"';
+			}
+			else if( elctn_cat === "legislation" ) {
+				aqString += ' @uri="dir=loi"';
+			}
+			else if( elctn_cat === "charg" ) {
+				aqString += ' @uri="section=charg"';
+			}
+			else if( elctn_cat === "ca" ) {
+				aqString += ' @uri="dir=ca"';
+			}
+			else if( elctn_cat === "un" ) {
+				aqString += ' @uri="dir=un"';
+			}
+			else if( elctn_cat === "pre" ) {
+				aqString += ' @uri="dir=pre-com"';
+			}
+			else if( elctn_cat === "spe" ) {
+				aqString += ' @uri="dir=spe-com"';
+			}
+			else if( elctn_cat === "rep" ) {
+				aqString += ' @uri="section=rep"';
+			}
 		}
-		else if( fqupdate === "datemodified_dt:[now-1year to now]" ) {
-			aqString += ' @date>today-365d';
+
+		if ( urlParams.filetype ) {
+			let filetype = urlParams.filetype.toLowerCase();
+			if ( filetype === "application/pdf" ) {
+				aqString += ' @filetype==(pdf)';
+			}
+			else if ( filetype === "ps" ) {
+				aqString += ' @filetype==(ps)';
+			}
+			else if ( filetype === "application/msword" ) {
+				aqString += ' @filetype==(doc,docx)';
+			}
+			else if ( filetype === "application/vnd.ms-excel" ) {
+				aqString += ' @filetype==(xls,xlsx)';
+			}
+			else if ( filetype === "application/vnd.ms-powerpoint" ) {
+				aqString += ' @filetype==(ppt,pptx)';
+			}
+			else if ( filetype === "application/rtf" ) {
+				aqString += ' @filetype==(rtf)';
+			}
```

Comment:
:fearful:

Do we need to support Rich Text Format documents? They probably have an HTML equivalent for WCAG compliance.